Ah, test_pickle fails:

```

ERROR: test_non_continuous_buffer (test.test_pickle.PyPicklingErrorTests.test_non_continuous_buffer) (proto=5)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/runner/work/cpython/cpython-ro-srcdir/Lib/test/pickletester.py", line 2330, in test_non_continuous_buffer
    self.dumps(pb, proto)
    ~~~~~~~~~~^^^^^^^^^^^
  File "/home/runner/work/cpython/cpython-ro-srcdir/Lib/test/test_pickle.py", line 66, in dumps
    p.dump(arg)
    ~~~~~~^^^^^
  File "/home/runner/work/cpython/cpython-ro-srcdir/Lib/pickle.py", line 499, in dump
    self.save(obj)
    ~~~~~~~~~^^^^^
  File "/home/runner/work/cpython/cpython-ro-srcdir/Lib/pickle.py", line 572, in save
    f(self, obj)  # Call unbound method with explicit self
    ~^^^^^^^^^^^
  File "/home/runner/work/cpython/cpython-ro-srcdir/Lib/pickle.py", line 896, in save_picklebuffer
    with obj.raw() as m:
         ~~~~~~~^^
BufferError: cannot extract raw buffer from non-contiguous buffer
```

--- a/Lib/test/pickletester.py
+++ b/Lib/test/pickletester.py
@@ -2326,7 +2326,7 @@ class AbstractPicklingErrorTests:
         for proto in protocols[5:]:
             with self.subTest(proto=proto):
                 pb = pickle.PickleBuffer(memoryview(b"foobar")[::2])
-                with self.assertRaises(pickle.PicklingError):
+                with self.assertRaises((pickle.PicklingError, BufferError)):
                     self.dumps(pb, proto)
 
     def test_buffer_callback_error(self):

